### PR TITLE
User service tests

### PR DESCRIPTION
**Pull request by Broseten**
Depends on #58, so merge it after merging that one.
Also please wait for other test to be merged to merge this one.

### CODE REVIEW

**Comment by Broseten** [comment]
Needs to fix two more tests - accept and reject mission (marked TODO)
edit: done

File: spaceMissionManager-service/src/test/java/cz/muni/fi/services/UserServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    private Map<Long, User> users = new HashMap<>();
+    private Map<Long, Mission> missions = new HashMap<>();
+
+    @BeforeClass
+    public void beforeClass() {
+        MockitoAnnotations.initMocks(this);
+
+        when(userDao.addUser(any(User.class))).then(invoke -> {
+            User user = invoke.getArgumentAt(0, User.class);
+
```
Comment (jsmadis) [inline spaceMissionManager-service/src/test/java/cz/muni/fi/services/UserServiceTest.java]: Potential null pointer exception if user is null

File: spaceMissionManager-service/src/test/java/cz/muni/fi/services/UserServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    @Mock
+    private MissionDao missionDao;
+
+    @Autowired
+    @InjectMocks
+    private UserService userService;
+
+    private Long idCounter;
+    private Map<Long, User> users = new HashMap<>();
+    private Map<Long, Mission> missions = new HashMap<>();
```
Comment (jsmadis) [inline spaceMissionManager-service/src/test/java/cz/muni/fi/services/UserServiceTest.java:46]: You never use this

EDIT: My bad, i get it now, this is fine

**Review by jsmadis — APPROVED** [review]

File: spaceMissionManager-service/src/test/java/cz/muni/fi/services/UserServiceTest.java (hunk 2 above)
Comment (Broseten) [reply spaceMissionManager-service/src/test/java/cz/muni/fi/services/UserServiceTest.java:46]: Can I ask what did you think is wrong?

**Review by mlynarikj — APPROVED** [review]

**Pull request merged**